### PR TITLE
Migrate GitRepositoryAdd to prefect

### PR DESCRIPTION
**Pull request by LucasG0**

### CODE REVIEW

File: backend/infrahub/git/tasks.py [hunk 1]
```diff
@@ -6,18 +6,50 @@
...
-from .models import RequestArtifactDefinitionGenerate, RequestArtifactGenerate
-from .repository import InfrahubRepository, get_initialized_repo
+from .models import GitRepositoryAdd, RequestArtifactDefinitionGenerate, RequestArtifactGenerate
 
 log = get_logger()
 
 
+@flow(name="git-repository-add-read-write")
+async def add_git_repository(model: GitRepositoryAdd) -> None:
+    log.info(
```
Comment (dgarros) [inline backend/infrahub/git/tasks.py]: I think we can remove this log it's redundant with what we are getting from prefect directly

**Review by dgarros — APPROVED** [review]

**Review by ajtmccarty — APPROVED** [review]

**Comment by ogenstad** [comment]
I would vote to enable the disabled tests related to proposed changes before merging this, i.e. these:

```python
    @pytest.mark.xfail(reason="FIXME Works locally but it's failling in Github Actions")
    async def test_happy_pipeline(self, db: InfrahubDatabase, happy_dataset: None, client: InfrahubClient) -> None:

    @pytest.mark.xfail(reason="FIXME Works locally but it's failling in Github Actions")
    async def test_conflict_pipeline(
```

Then we need to figure out what's required to actually trigger the jobs to be executed in serial, so that the testing experience is somewhat similar to what we had before. In this PR we see it in the failing integration test where some more work will need to be done, but it could also be other things that are required with regards to these disabled tests. It might be that we need to rewrite them from scratch depending on the options we have with prefect. But if we look at this change:

```python
            if context.service:
                context.service.workflow.submit_workflow(
                    workflow=GIT_REPOSITORY_ADD, parameters={"model": git_repo_add_model}
                )
```

We need the submit_workflow method to actually run the import repo function in the background and add the changes back to the current test instance in the same way as the old RabbitMQ message did.

**Comment by dgarros** [comment]
> We need the submit_workflow method to actually run the import repo function in the background and add the changes back to the current test instance in the same way as the old RabbitMQ message did.

That should already work as expected when the tests are using `WorkflowLocalExecution`

**Comment by ogenstad** [comment]
> That should already work as expected when the tests are using `WorkflowLocalExecution`

Ah, right so will just need to update that integration test. I still think it would be good to have the other tests enabled again though.

**Comment by dgarros** [comment]
@LucasG0 FYI I rebased this branch after the latest changes in `develop` to see if resolving the merge conflicts would be enough to fix the tests ... it was not, looks like some tests still need to be fixed

File: backend/tests/integration/proposed_change/test_proposed_change_conflict.py [hunk 2]
```diff
@@ -10,21 +13,37 @@
...
 
 
 class TestProposedChangePipelineConflict(TestInfrahubApp):
+    @pytest.fixture(scope="class")
+    def car_dealership_copy(self):
+        """
+        Copies car-dealership local repository to a temporary folder, with a new name.
+        This is needed for this test as using car-dealership folder leads to issues most probably
+        related to https://github.com/opsmill/infrahub/issues/4296 as some other tests use this same repository.
+        """
```
Comment (LucasG0) [inline backend/tests/integration/proposed_change/test_proposed_change_conflict.py:36]: #4296 for reference

**Pull request merged**